packages/display/src/DisplayObject.ts

```diff
@@ -550,6 +550,8 @@ export abstract class DisplayObject extends utils.EventEmitter<DisplayObjectEven
         const parentRef = this.parent;
 
         this.parent = null;
+        // inherit the worldAlpha of the parent node
+        this._tempDisplayObjectParent.worldAlpha = parentRef?.worldAlpha || 1;
```

Comment:
Minor suggestion to handle 0

```suggestion
        this._tempDisplayObjectParent.worldAlpha = parentRef?.worldAlpha ?? 1;
```